Sources/Build/BuildPlan.swift

```diff
@@ -657,24 +683,45 @@ public final class SwiftTargetBuildDescription {
         self.tempsPath = buildParameters.buildPath.appending(component: target.c99name + ".build")
         self.derivedSources = Sources(paths: [], root: tempsPath.appending(component: "DerivedSources"))
         self.pluginDerivedSources = Sources(paths: [], root: buildParameters.dataPath)
+        self.pluginDerivedResources = []
         self.buildToolPluginInvocationResults = buildToolPluginInvocationResults
         self.prebuildCommandResults = prebuildCommandResults
 
+        func handlePluginDerivedFile(path absPath: AbsolutePath) {
+            // FIXME: Carry `additionalFileRules` until here.
+            let rule = TargetSourcesBuilder.computeRule(for: absPath, toolsVersion: toolsVersion, additionalFileRules: [])
+            print("Determined \(rule) for generated file at \(absPath)")
+            switch rule {
+            case .compile:
+                if absPath.extension == "swift" {
+                    let relPath = absPath.relative(to: self.pluginDerivedSources.root)
+                    self.pluginDerivedSources.relativePaths.append(relPath)
+                } else {
+                    // FIXME: Error for unsupported source file type.
+                }
+            case .copy, .processResource:
+                if let resource = TargetSourcesBuilder.resource(for: absPath, with: rule) {
+                    self.pluginDerivedResources.append(resource)
+                } else {
+                    // FIXME: Error for not obtaining a `Resource` object for a file.
```

Comment:
should this throw / warn?

Reply:
I changed it to be ignored since it looks like `TargetSourcesBuilder` itself would already warn in this case.